src/placeos-models/user_jwt.cr

```diff
@@ -68,30 +68,31 @@ module PlaceOS::Model
       scope.includes?(PUBLIC)
     end
 
-    def get_scope(scope_name : String)
-      scope.each do |entry|
-        if entry.resource == scope_name
-          return entry.access
-        end
+    def get_access(scope_name : String) : Scope::Access
+      existing = scope.find &.resource.==(scope_name)
+
+      if existing.nil?
+        Log.warn { "scope #{scope_name} not present in JWT" }
+        Access::None
+      else
+        existing.access
       end
-      Log.warn { {message: "unknown scope #{scope_name}"} }
-      raise Error::NoScope.new
     end
 
     struct Scope
       @[Flags]
       enum Access
         Read
         Write
-        Full
       end
 
-      getter resource : String
+      getter resource : (String)
 
       getter access : Access
 
       def initialize(@resource, access : Access? = nil)
         access = Access::Full if access.nil?
+        access = Access::None if @resource == "guest"
```

Comment:
@stakach and `public`?

Reply:
Doesn't public mean Full access?

Reply:
public means full access, guest scope doesn't mean no access
guest scope should be treated like a normal scope, with only public being special case

Reply:
I think `public` is slightly confusing in that case, I think it would make more sense as something like `global`.

I don’t think a change would be too hard as we’re not checking for `public` scope anywhere (at least on API)

Reply:
all existing installations are configured to use public though, and it's the default provided by doorkeeper
https://github.com/PlaceOS/auth/blob/master/config/initializers/doorkeeper.rb#L82

so would be a breaking change for existing deployments - so I would recommend not changing it for historical reasons
the scope name is meant to represent access to all public APIs

Reply:
Sweet, makes sense.

I guess `PUBLIC="public"` and `GUEST="guest"` should be constants in the class